server/src/main/java/org/opensearch/common/compress/NullCompressor.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.compress;
+
+import org.opensearch.common.bytes.BytesReference;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
+ * {@link Compressor} no compressor implementation.
+ *
+ * @opensearch.internal
+ */
+public class NullCompressor implements Compressor {
```

Comment:
`CompressorFactory.compressor` is nullable, why do we need an explicit null compressor? Can we just check `compressor == null ? indexOutputOutputStream : compressor.threadLocalOutputStream(indexOutputOutputStream)` and do away with this empty class?

Reply:
TBH it is just a habit to use Null class and do not check it on null every where, but I think with None it would be better